src/platform/backends/qemu/firewall_config.cpp

```diff
@@ -220,54 +236,112 @@ void clear_iptables_rules_for(const QString& table, const QString& bridge_name,
             // Remove the policy type since delete doesn't use that
             rule.remove(0, 3);
 
-            // Pass the chain and rule wholesale since we capture the whole line
-            delete_iptables_rule(table, QStringList() << rule);
+            try
+            {
+                // Pass the chain and rule wholesale since we capture the whole line
+                delete_firewall_rule(firewall, table, QStringList() << rule);
+            }
+            catch (const mp::FirewallException& e)
+            {
+                mpl::log(mpl::Level::error, category,
+                         fmt::format("Error deleting firewall rule '{}': {}", rule, e.what()));
+            }
         }
     }
 }
-} // namespace
 
-mp::IPTablesConfig::IPTablesConfig(const QString& bridge_name, const std::string& subnet)
-    : bridge_name{bridge_name},
-      cidr{QString("%1.0/24").arg(QString::fromStdString(subnet))},
-      comment{multipass_iptables_comment(bridge_name)}
+bool is_firewall_in_use(const QString& firewall)
 {
+
+    return std::any_of(firewall_tables.cbegin(), firewall_tables.cend(), [&firewall](const QString& table) {
+        QRegularExpression re{"^-[ARIN]"};
+        auto rule_lines = get_firewall_rules(firewall, table).split('\n');
+
+        return std::any_of(rule_lines.cbegin(), rule_lines.cend(),
+                           [&re](const QString& line) { return re.match(line).hasMatch(); });
+    });
+}
+
+// We require a >= 5.2 kernel to avoid weird conflicts with xtables and support for inet table NAT rules.
+// Taken from LXD :)
+bool kernel_supports_nftables()
+{
+    const auto kernel_version{MP_UTILS.get_kernel_version()};
     try
     {
-        clear_all_iptables_rules();
-        set_iptables_rules(bridge_name, cidr, comment);
+        auto kernel_supported{version::Semver200_version(kernel_version) >= version::Semver200_version("5.2.0")};
```

Comment:
Right, but again, following what ufw and LXD did, they ran into issues with older kernels and made their minimums 5.0 and 5.2 respectively.  Since we kind of follow how LXD does things wrt networking, I think making it 5.2 is the safest to avoid any unnecessary issues for us.

Reply:
OK cool.